### ISSUE TITLE
add index to delayWhen selector

### ISSUE THREAD

**Issue by Dorus** [ReactiveX/rxjs#2388]
Most operators include the index of the current element in the selector function For example `lter(predicate: function(value: T, **index**: number)` and ` mergeMap(project: function(value: T, **?index**: number)`.

`delayWhen` however is missing the index:
```
public delayWhen(delayDurationSelector: function(value: T): Observable, subscriptionDelay: Observable): Observable
```
However, it would be especially useful if `delayDurationSelector` included the index. That would make it easy to create an incremental delay, for example when `delayWhen` is combined with `retryWhen`.

**Comment by macklinu** [comment]
Was this resolved by #2473? Anything else left to do here?

**Comment by cartant** [comment]
> Was this resolved by #2473?

That seems to be the case, yes.
